app/assets/javascripts/backbone/views/calendar_step_modal_view.coffee

```diff
@@ -1,22 +1,32 @@
 class MS.Views.CalendarStepModalView extends Backbone.View
 
-  el: '.calendar'
+  el: '#calendar-step-modal'
+
+  initialize: (params) ->
+    @context = params.context
+    if params.id
+      @id = params.id
+      @showStep()
+    else
+      @loanId = params.loanId
+      @showNewStep()
 
   events:
-    'click .loan-calendar .cal-step': 'showStepModal'
-    'click a.action-delete': 'deleteStep'
+    'click a.action-delete': 'hideModal'
+    'click #new_project_step .cancel': 'hideModal'
+
+  showStep: ->
+    $.get '/admin/project_steps/' + @id, context: @context, (html) =>
+      @replaceContent(html)
 
-  showStepModal: (e) ->
-    calStep = e.currentTarget
-    id = @$(calStep).attr('data-step-id')
-    MS.loadingIndicator.show()
-    $.get '/admin/project_steps/' + id, context: 'calendar', (html) =>
+  showNewStep: ->
+    $.get '/admin/project_steps/new', context: @context, loan_id: @loanId, (html) =>
       @replaceContent(html)
 
   replaceContent: (html) ->
-    @$('#calendar-step-modal').find('.modal-content').html(html)
-    @$('#calendar-step-modal').modal({show: true})
+    @$el.find('.modal-content').html(html)
+    @$el.modal({show: true})
```

Comment:
Are you sure? Can you link to where it says that in the docs? I am
confused...

On Mon, May 16, 2016 at 11:58 AM, Elizabeth Mitchell <
notifications@github.com> wrote:

> In app/assets/javascripts/backbone/views/calendar_step_modal_view.coffee
> <https://github.com/sassafrastech/madeline_system/pull/78#discussion_r63377119>
> :
>
> >        @replaceContent(html)
> >
> >    replaceContent: (html) ->
> > -    @$('#calendar-step-modal').find('.modal-content').html(html)
> > -    @$('#calendar-step-modal').modal({show: true})
> > +    @$el.find('.modal-content').html(html)
> > +    @$el.modal({show: true})
>
> ('show') is shorthand. I changes it to that, as I think that is best.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/sassafrastech/madeline_system/pull/78/files/3624899d75a6267d9e09080d65279d58710f6b38#r63377119>
>



-- 
Tom Smyth

Worker-Owner, Sassafras Tech Collective
Specializing in innovative, usable tech for social change
sassafras.coop *·* @sassafrastech

Resident, Touchstone Cohousing
touchstonecohousing.org